### DISCUSSION TITLE
Find word not working

### DISCUSSION THREAD

**Discussion by Dennis-Zhang-SH** [ayamir/nvimdots#579] [Q&A] [answered]
Hi, I am a newbie to vim and neovim, I was using vs code mostly in the past, and I just began to use this neovim setting, it's blazing fast and easy to use, but I wonder why that `space f f` works fine on searching files, but `space f w` failed to search and return with a warning.

Directly using grep_string return with an not found libtelescope.so error, should I install it?

**Comment by charliie-dev** [comment]
I think `telescope.nvim` might not install successfully. I would suggest remove it from the plugin folder and re-open nvim to let `lazy.nvim` re-install it again.
If the error persist, show us some screenshots and the error log.

**Comment by Dennis-Zhang-SH** [comment]
Hi I reinstall it, and the result is in the following:

`space fw` not working

![image](https://user-images.githubusercontent.com/51769674/225808247-cb0775c5-b938-42a0-b2b1-31e377e6be37.png)

`space ff` works fine

![image](https://user-images.githubusercontent.com/51769674/225808387-deeed147-7504-4066-bdbc-318641158f29.png)

**Comment by Jint-lzxy** [comment, marked as answer]
Hi @Dennis-Zhang-SH first you need to install a [patched font](https://www.nerdfonts.com/font-downloads) in order to render the icons correctly.

About your question, do you have all the necessary dependencies installed according to [Wiki:Prerequisites#required-packages-installation](https://github.com/ayamir/nvimdots/wiki/Prerequisites#required-packages-installation-for-archlinux)? [BurntSushi/ripgrep](https://github.com/BurntSushi/ripgrep) is required for `live_grep` and `grep_string`.

**Reply by aarnphm** [reply]
I think this has to do with live_grep_args. `Telescope live_grep` should work as usual

**Reply by Jint-lzxy** [reply]
Hmm but AFAIU any feature that requires a recursive search in directories for a regex pattern requires `ripgrep` - ref: [telescope.nvim#suggested-dependencies](https://github.com/nvim-telescope/telescope.nvim#suggested-dependencies)

**Reply by aarnphm** [reply]
I have rg, and i did run into this some of the times.

**Reply by Dennis-Zhang-SH** [reply]
> Hi @Dennis-Zhang-SH first you need to install a [patched font](https://www.nerdfonts.com/font-downloads) in order to render the icons correctly.
> 
> About your question, do you have all the necessary dependencies installed according to [Wiki:Prerequisites#required-packages-installation](https://github.com/ayamir/nvimdots/wiki/Prerequisites#required-packages-installation-for-archlinux)? [BurntSushi/ripgrep](https://github.com/BurntSushi/ripgrep) is required for `live_grep` and `grep_string`.

Hi, I installed ripgrep and it works, but it didn't work on the nerd font, I have installed it on ~/.local/share/fonts.

![image](https://user-images.githubusercontent.com/51769674/226230997-d0f3769a-e2f7-465a-8f3e-390ca4e250aa.png)

![image](https://user-images.githubusercontent.com/51769674/226231053-768af3f9-f930-43b0-aa44-4a33dceced57.png)

Besides, what's this error? Each time I got it when I open up a terminal but couldn't know what is it.

![image](https://user-images.githubusercontent.com/51769674/226230875-5833ce98-faed-401e-84f6-ce875e0dc786.png)

**Reply by charliie-dev** [reply]
@Dennis-Zhang-SH For rebuilding the fonts, try `fc-cache -f -v`, might need `sudo`.

**Reply by Dennis-Zhang-SH** [reply]
> @Dennis-Zhang-SH For rebuilding the fonts, try `fc-cache -f -v`, might need `sudo`.

I tried rebuilding it, not working still

**Reply by charliie-dev** [reply]
wait a sec. you're using a wsl right? then u also need to install the font on windows and select the font in windows powershell.

**Reply by Dennis-Zhang-SH** [reply]
> wait a sec. you're using a wsl right? then u also need to install the font on windows and select the font in windows powershell.

I fix this by configuring the windows terminal, thanks anyway.

**Reply by charliie-dev** [reply]
> Besides, what's this error? Each time I got it when I open up a terminal but couldn't know what is it.

![圖片](https://user-images.githubusercontent.com/32497323/226234308-020ae6ec-06f1-49bd-bff9-8ba24a20b974.png)
use this to display all messages.

**Reply by Dennis-Zhang-SH** [reply]
> > Besides, what's this error? Each time I got it when I open up a terminal but couldn't know what is it.
> 
> ![圖片](https://user-images.githubusercontent.com/32497323/226234308-020ae6ec-06f1-49bd-bff9-8ba24a20b974.png) use this to display all messages.

I got the following message:

Error executing vim.schedule lua callback: .../nvim-tree.lua/lua/nvim-tree/actions/root/change-dir.lua:50: Vim(lcd):E472: Command failed
stack traceback:
        [C]: in function 'cmd'
        .../nvim-tree.lua/lua/nvim-tree/actions/root/change-dir.lua:50: in function 'cd'
        .../nvim-tree.lua/lua/nvim-tree/actions/root/change-dir.lua:67: in function 'f'
        .../nvim-tree.lua/lua/nvim-tree/actions/root/change-dir.lua:60: in function 'force_dirchange'
        .../nvim-tree.lua/lua/nvim-tree/actions/root/change-dir.lua:46: in function 'fn'
        ...cal/share/nvim/site/lazy/nvim-tree.lua/lua/nvim-tree.lua:74: in function 'change_root'
        ...y/nvim-tree.lua/lua/nvim-tree/actions/tree/find-file.lua:59: in function 'fn'
        ...cal/share/nvim/site/lazy/nvim-tree.lua/lua/nvim-tree.lua:375: in function 'callback'
        ...are/nvim/site/lazy/nvim-tree.lua/lua/nvim-tree/utils.lua:336: in function <...are/nvim/site/lazy/nvim-tree.lua/lua/nvim-tree/utils.lua:335>

**Reply by charliie-dev** [reply]
Hmmm, this seems like a upstream bug, related to `nvim-tree` and https://github.com/ahmedkhalf/project.nvim.
I think this should be filed in new issue in this repo, maybe u can search for upstream repos to find some useful infos if possible.

**Reply by Dennis-Zhang-SH** [reply]
> Command failed

Sure, one last question, how could I find out the lsp prompt? Like in the vscode, you could always navigate to problems tab in the terminal window, which would be very convenient.

**Reply by charliie-dev** [reply]
I not sure what u mean, I haven't use vscode for coding in a very long time. But maybe this is what u need?

https://github.com/ayamir/nvimdots/blob/ed54fd687d7347168b887b8b420d42f520f8b5d6/lua/keymap/completion.lua#L12-L13

or

https://github.com/ayamir/nvimdots/blob/ed54fd687d7347168b887b8b420d42f520f8b5d6/lua/keymap/tool.lua#L84-L102

**Reply by charliie-dev** [reply]
Personally I don't think it's necessary to replicate every function in vscode, we can just done the same thing in different way. So maybe next time u can describe what exactly u want to achieve instead of describing a function in vscode. Abstract the concept.

**Reply by Dennis-Zhang-SH** [reply]
> Personally I don't think it's necessary to replicate every function in vscode, we can just done the same thing in different way. So maybe next time u can describe what exactly u want to achieve instead of describing a function in vscode. Abstract the concept.

Oh it's the `<leader>tw`, thanks, sure I would describe the concept next time, I am too familiar with vscode, sorry about that.

**Comment by Dennis-Zhang-SH** [comment]
Thanks for all the help, I think I could start using my neovim journey :).
